### PR TITLE
[expo-permissions] Fix `permission cannot be null or empty` in the bare workflow

### LINKED ISSUE #8909: [expo-permissions] Getting native exception: permission cannot be null or empty
Follow up to the https://github.com/expo/expo/issues/5548#issuecomment-647359064.

This code:
```ts
import {Permissions} from 'react-native-unimodules'
const {granted} = await Permissions.askAsync(
    Permissions.USER_FACING_NOTIFICATIONS,
    Permissions.NOTIFICATIONS,
  )
```
in the bare workflow on Android will reject with exception:
`[Mon Jun 22 2020 10:00:20.368] WARN 0 [Error: Encountered an exception while calling native method: Exception occurred while executing exported method askAsync on module ExpoPermissions: permission cannot be null or empty]`

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Fixes https://github.com/expo/expo/issues/8909
# How

Some permissions like `NOTIFICATIONS` or `USER_FACING_NOTIFICATIONS` aren't supported by Android. So, we shouldn't ask for them - we can resolve without asking the system about them.

# Test Plan

- NCL ✅

# Changelog 

- Fixed `askAsync` rejecting with `permission cannot be null or empty` in the bare workflow.

### CODE REVIEW

**Review by mczernek — APPROVED** [review]
I generally approve, but would reconsider if adding such wrapper, apart from reducing duplication (which is great), does not include some complication and reduce readibility.

**Comment by lukmccall** [comment]
> I generally approve, but would reconsider if adding such wrapper, apart from reducing duplication (which is great), does not include some complication and reduce readibility.

I believe that this wrapper doesn't add too many complications, because it doesn't od anything complex. What's more, it's hide parsing code from `getAsync` and `askAsync` methods.

**Comment by mczernek** [comment]
> > I generally approve, but would reconsider if adding such wrapper, apart from reducing duplication (which is great), does not include some complication and reduce readibility.
> 
> I believe that this wrapper doesn't add too many complications, because it doesn't od anything complex. What's more, it's hide parsing code from `getAsync` and `askAsync` methods.

I will leave it to your gut, so feel free to merge it as it is :)

That was exactly my point. It does not do anything complex but makes it a little bit obfuscated, since its name does not describe its purpose clearly and indicated that it does something complex.

**Pull request merged**
